ttnn/cpp/ttnn/operations/data_movement/untilize/device/untilize_program_factory.cpp

```diff
@@ -146,8 +169,8 @@ operation::ProgramWithCallbacks untilize_multi_core(
                 all_cores,
                 WriterDataMovementConfig(writer_ct_args));
         } else {
-            bool stick_size_is_power_of_two = is_power_of_two_at_least_32(block_size_nbytes);
-            uint32_t log2_stick_size = stick_size_is_power_of_two ? (std::uint32_t)std::log2(block_size_nbytes) : 0;
+            bool stick_size_is_power_of_two = is_power_of_two_at_least_32(stick_size);
+            uint32_t log2_stick_size = stick_size_is_power_of_two ? (std::uint32_t)std::log2(stick_size) : 0;
```

Comment:
btw std::bit_width(stick_size) - 1 should be equal to the (std::uint32_t)std::log2(stick_size)
https://en.cppreference.com/w/cpp/numeric/bit_width